### PR TITLE
Add linter setups

### PR DESCRIPTION
**Pull request by wesrich**
Not sure if we would prefer to break this into multiple files (especially since some parts of this are Rails-specific, Ruby-specific, or may be applied across other projects).

Open to suggestions on the possibility of splitting it, or other content suggestions.

### CODE REVIEW

File: rails/linter-config.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
+# Using Code Linters on Rails
```
Comment (calebwoods) [inline rails/linter-config.md]: I could see breaking these into their own sub category of Linters and Auditors.  Seems like a number of these will apply to all projects Scss, and ESLint specifically

File: rails/linter-config.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+```bash
+bundle exec rubocop
+```
+
+You can also tell Semaphore to only exit with a failing error code on ERROR and
+above by specifying a `--fail-level` argument:
+
+```bash
+bundle exec rubocop --fail-level ERROR
```
Comment (calebwoods) [inline rails/linter-config.md]: Should we talk about TODO files with Rubocop?  Could make sense if we break it out into a separate entry

File: rails/linter-config.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+```
+
+## ESLint
+* [ESLint Documentation](http://eslint.org/docs/user-guide)
+
+### Create the config file
+You'll want to copy these two files into your project directory:
+* [.eslintrc.js](https://github.com/RoleModel/almanac/blob/add_rubocop_scss_and_eslint/.eslintrc.js)
```
Comment (calebwoods) [inline rails/linter-config.md]: These are internal files.  Are there any public copies?

**Comment by calebwoods** [comment]
One standard that I've started to think about is moving tasks in Semaphore to have separate threads for Test and Linters/Security Checks.

screen shot 2016-08-26 at 10 49 51 am

**Comment by wesrich** [comment]
Separate threads for security and linters is an interesting idea ...

One reason I've seen to do Rubocop (other linters may be applicable) during Semaphore's shared "setup" phase, is because it fails fast in the case of a `binding.pry` or syntax that's invalid. In those cases, you wouldn't wait around for a 20-30 minute test suite only to find out 25 minutes in that there's a major syntax issue.

File: rails/linter-config.md (hunk 3 above)
Comment (wesrich) [inline rails/linter-config.md]: Whoops! Yes, the public copies should be in the Rails Template. I'll update the links shortly. (The files weren't in Rails Template when Ben and I were writing the first draft of this document.)

Thanks for pointing this out. :)

**Pull request merged**